src/server/set_family.cc

```diff
@@ -504,6 +505,8 @@ OpResult<SvArray> InterResultVec(const ResultStringVec& result_vec, unsigned req
 
   for (const auto& k_v : uniques) {
     if (k_v.second == required_shard_cnt) {
+      if (limit != 0 && result.size() >= limit)
+        return result;
       result.push_back(k_v.first);
```

Comment:
You probably want to do this after the push_back, so you would return immediately on reaching the limit